liberty-maven-plugin/src/main/java/net/wasdev/wlp/maven/plugins/ServerXmlDocument.java

```diff
@@ -0,0 +1,159 @@
+/**
+ * (C) Copyright IBM Corporation 2017.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package net.wasdev.wlp.maven.plugins;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileWriter;
+import java.io.InputStream;
+import java.net.URL;
+import java.net.URLConnection;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.transform.OutputKeys;
+import javax.xml.transform.Transformer;
+import javax.xml.transform.TransformerFactory;
+import javax.xml.transform.dom.DOMSource;
+import javax.xml.transform.stream.StreamResult;
+
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+
+public class ServerXmlDocument {
+    
+    private ServerXmlDocument() {    
+    }
+    
+    public static boolean isFoundTagName(String fileName, String tag) throws Exception {
+
+        DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
+        DocumentBuilder builder = factory.newDocumentBuilder();
+        Document doc = builder.parse(fileName);
+        
+        NodeList appList = doc.getElementsByTagName(tag);
+        
+        return appList != null && appList.getLength() > 0;
+    }
+    
+    public static boolean isFoundWebApplication(String filePath, boolean bTarget) throws Exception {
```

Comment:
I only see checking the location value ended with ".war" in the method but not the filename. Sometime, the location value (e.g. in the ferret sample) is a system property.

https://github.com/WASdev/sample.ferret/blob/master/src/main/wlp/server.xml

```xml
<application name="ferret" context-root="/ferret" location="${wlp.install.dir}/../../${appLocation}" type="war">
```

```xml
    public static boolean isFoundWebAppElement(NodeList nodeList) {
         
        boolean bFound = false; 
         
        for(int i = 0; i < nodeList.getLength(); i++) {
            Node typeNode = nodeList.item(i).getAttributes().getNamedItem("type");
            if (typeNode != null) {
                String typeValue = typeNode.getNodeValue();
                
                if (typeValue != null && typeValue.equals("war")) {
                    bFound = true;
                    break;
                }
            }
    
            Node locNode = nodeList.item(i).getAttributes().getNamedItem("location");
            if (locNode != null) {
                String locValue = locNode.getNodeValue();
                
                if (locValue != null && locValue.endsWith(".war")) {
                    bFound = true;
                    break;
                }
            }
        }
        return bFound;
    }
```